routing/graph.go

```diff
@@ -1,4 +1,101 @@
 package routing
 
-// TODO(roasbeef): abstract out graph to interface
-//  * add in-memory version of graph for tests
+import (
+	"github.com/coreos/bbolt"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing/route"
+)
+
+// routingGraph is an abstract interface that provides information about nodes
+// and edges to pathfinding.
+type routingGraph interface {
+	// forEachNodeChannel calls the callback for every channel of the given node.
+	forEachNodeChannel(nodePub route.Vertex,
```

Comment:
Suggestion to further decouple this from channeldb: figure out what information from `ChannelEdgeInfo` and `ChannelEdgePolicy` we need during path finding, and add that to a new type that can be part of this interface.

Reply:
I tried it out. It is possible, but there are multiple ways to shape the data structures. I think it is better to leave it as is for now, also because it enlarges the scope of the pr without a direct benefit, and revisit when in-memory pathfinding is implemented. It will then probably be more clear what the best structure is.